File: src/mail/mail.service.ts
import { Injectable, Logger } from '@nestjs/common';
import { ConfigService } from '@nestjs/config';
import * as nodemailer from 'nodemailer';

@Injectable()
export class MailService {
  private readonly logger = new Logger(MailService.name);
  private transporter: nodemailer.Transporter;

  constructor(private config: ConfigService) {
    const host = this.config.get('MAIL_HOST');
    const user = this.config.get('MAIL_USER');

    if (host && user) {
      this.transporter = nodemailer.createTransport({
        host,
        port: this.config.get<number>('MAIL_PORT', 587),
        secure: false,
        auth: { user, pass: this.config.get('MAIL_PASS') },
      });
    }
  }

  async sendNewLeadNotification(lead: { name: string; phone: string; email?: string; company?: string; message?: string; service?: string }) {
    if (!this.transporter) {
      this.logger.warn('Mail transporter not configured, skipping email');
      return;
    }

    try {
      await this.transporter.sendMail({
        from: this.config.get('MAIL_FROM'),
        to: this.config.get('MAIL_ADMIN'),
        subject: `[GESE] Lead mới: ${lead.name} - ${lead.phone}`,
        html: `
          <h2>Lead mới từ website GESE</h2>
          <table border="1" cellpadding="8" cellspacing="0" style="border-collapse:collapse">
            <tr><td><strong>Họ tên</strong></td><td>${lead.name}</td></tr>
            <tr><td><strong>SĐT</strong></td><td>${lead.phone}</td></tr>
            <tr><td><strong>Email</strong></td><td>${lead.email || '-'}</td></tr>
            <tr><td><strong>Công ty</strong></td><td>${lead.company || '-'}</td></tr>
            <tr><td><strong>Dịch vụ</strong></td><td>${lead.service || '-'}</td></tr>
            <tr><td><strong>Nội dung</strong></td><td>${lead.message || '-'}</td></tr>
          </table>
        `,
      });
      this.logger.log(`Lead notification sent for ${lead.name}`);
    } catch (err) {
      this.logger.error('Failed to send lead notification', err);
    }
  }
}

File: api/src/main/java/com/gese/firesecurity/mail/MailService.java
package com.gese.firesecurity.mail;

import com.gese.firesecurity.lead.dto.CreateLeadRequest;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.mail.javamail.JavaMailSender;
import org.springframework.mail.javamail.MimeMessageHelper;
import org.springframework.scheduling.annotation.Async;
import org.springframework.stereotype.Service;

@Slf4j
@Service
@RequiredArgsConstructor
public class MailService {

    private final JavaMailSender mailSender;

    @Value("${app.mail.from:}")
    private String mailFrom;

    @Value("${app.mail.admin:}")
    private String mailAdmin;

    @Async
    public void sendNewLeadNotification(CreateLeadRequest lead) {
        if (mailFrom.isBlank() || mailAdmin.isBlank()) {
            log.warn("Mail not configured, skipping notification");
            return;
        }

        try {
            var message = mailSender.createMimeMessage();
            var helper = new MimeMessageHelper(message, true, "UTF-8");
            helper.setFrom(mailFrom);
            helper.setTo(mailAdmin);
            helper.setSubject("[GESE] Lead mới: " + lead.getName() + " - " + lead.getPhone());
            helper.setText(String.format("""
                <h2>Lead mới từ website GESE</h2>
                <table border="1" cellpadding="8" cellspacing="0" style="border-collapse:collapse">
                  <tr><td><strong>Họ tên</strong></td><td>%s</td></tr>
                  <tr><td><strong>SĐT</strong></td><td>%s</td></tr>
                  <tr><td><strong>Email</strong></td><td>%s</td></tr>
                  <tr><td><strong>Công ty</strong></td><td>%s</td></tr>
                  <tr><td><strong>Dịch vụ</strong></td><td>%s</td></tr>
                  <tr><td><strong>Nội dung</strong></td><td>%s</td></tr>
                </table>
                """,
                lead.getName(), lead.getPhone(),
                lead.getEmail() != null ? lead.getEmail() : "-",
                lead.getCompany() != null ? lead.getCompany() : "-",
                lead.getService() != null ? lead.getService() : "-",
                lead.getMessage() != null ? lead.getMessage() : "-"
            ), true);

            mailSender.send(message);
            log.info("Lead notification sent for {}", lead.getName());
        } catch (Exception e) {
            log.error("Failed to send lead notification", e);
        }
    }
}
